core/actions/_lib/zodTypes.ts

```diff
@@ -32,5 +32,27 @@ class StringWithTokens extends z.ZodString {
 	}
 }
 
+const actionInstanceShape = {
+	name: z.string(),
+	description: z.string(),
+	icon: z.string(),
+	action: z.string(),
+	actionInstanceId: z.string().uuid(),
+};
+
+export type ActionInstanceConfig = z.infer<z.ZodObject<typeof actionInstanceShape>>;
+
+// @ts-expect-error FIXME: '{ name: z.ZodString; description: z.ZodString; icon: z.ZodString; action: z.ZodString; actionInstanceId: z.ZodString; }' is assignable to the constraint of type 'T_1', but 'T_1' could be instantiated with a different subtype of constraint 'ZodRawShape'.ts(2417)
+class ActionInstance extends z.ZodObject<typeof actionInstanceShape, "strip", z.ZodTypeAny> {
+	static create = () =>
+		new ActionInstance({
+			typeName: "ActionInstance" as z.ZodFirstPartyTypeKind.ZodObject,
+			shape: () => actionInstanceShape,
+			catchall: z.never(),
+			unknownKeys: "strip",
+		});
+}
```

Comment:
i initially went a different route with the rule config, treating the action config the same as eg the config for `pubEnteredStage`.

I can still see us referring to other actions when configuring actions, so i decided to leave it in. please let me know if i should just remove it! this also goes for the `actionInstance` context in `ui`